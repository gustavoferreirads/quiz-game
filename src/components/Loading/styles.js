import { StyleSheet } from 'react-native';
import metrics from '../../styles/metrics';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },
  horizontal: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: metrics.size,
  },
});
