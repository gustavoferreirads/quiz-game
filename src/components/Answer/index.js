import React from 'react';
import * as PropTypes from 'prop-types';
import { Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

const Answer = ({ option, text, onPress }) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <View style={styles.contentOption}>
      <Text style={styles.option}>{option}</Text>
    </View>
    <Text style={styles.text}>{text}</Text>
  </TouchableOpacity>
);

Answer.defaultProps = {};
Answer.propTypes = {
  option: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default Answer;
