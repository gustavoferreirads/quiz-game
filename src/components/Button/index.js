import React, { Fragment } from 'react';
import * as PropTypes from 'prop-types';
import { Text, TouchableOpacity } from 'react-native';
import styles from './styles';

const Button = ({ style, children, type, onPress }) => (
  <TouchableOpacity style={[styles.container, style, styles[`button-${type}`]]} onPress={onPress}>
    <Text style={[styles.text, styles[`text-${type}`]]}>{children}</Text>
  </TouchableOpacity>
);

Button.defaultProps = {
  type: 'inline',
  onPress: () => {},
  style: {},
  children: <Fragment />,
};

Button.propTypes = {
  children: PropTypes.element,
  style: PropTypes.object,
  type: PropTypes.string,
  onPress: PropTypes.func,
};

export default Button;
