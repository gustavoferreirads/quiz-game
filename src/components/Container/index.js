import React from 'react';
import { View } from 'react-native';
import * as PropTypes from 'prop-types';

import styles from './styles';

const Container = ({ children, style }) => (
  <View style={[styles.container, style]}>{children}</View>
);
Container.defaultProps = {
  style: {},
};
Container.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.array]).isRequired,
  style: PropTypes.object,
};

export default Container;
