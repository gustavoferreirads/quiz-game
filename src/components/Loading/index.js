import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator, View } from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import { colors } from '../../styles';

export const withLoading = Component => {
  const WithLoading = ({ questions, ...props }) =>
    questions.length > 0 ? (
      <Component {...props} />
    ) : (
      <View style={[styles.container, styles.horizontal]}>
        <ActivityIndicator size={0} color={colors.primary} />
      </View>
    );

  WithLoading.propTypes = {
    questions: PropTypes.arrayOf(PropTypes.object).isRequired,
  };

  return connect(({ Quiz }) => ({ questions: Quiz.questions }))(WithLoading);
};
