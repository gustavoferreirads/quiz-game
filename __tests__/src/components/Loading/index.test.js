import React from 'react';
import { Provider } from 'react-redux';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { withLoading } from '../../../../src/components/Loading';

describe('Loading HOC', () => {
  it('renders a internal Component', () => {
    const mockStore = configureStore({ Quiz: { questions: [{ id: 1 }] } });
    const Child = withLoading(View);
    const rendered = shallow(
      <Provider store={mockStore}>
        <Child />
      </Provider>,
    );
    expect(rendered).toMatchSnapshot();
  });

  it('renders a Loading ', () => {
    const mockStore = configureStore({ Quiz: { questions: [] } });
    const Child = withLoading(View);
    expect(
      shallow(
        <Provider store={mockStore}>
          <Child />
        </Provider>,
      ),
    ).toMatchSnapshot();
  });
});
