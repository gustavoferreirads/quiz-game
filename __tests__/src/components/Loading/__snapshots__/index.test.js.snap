// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading HOC renders a Loading  1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "Quiz": Object {
          "questions": Array [],
        },
      },
    }
  }
>
  <Connect(WithLoading) />
</ContextProvider>
`;

exports[`Loading HOC renders a internal Component 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "Quiz": Object {
          "questions": Array [
            Object {
              "id": 1,
            },
          ],
        },
      },
    }
  }
>
  <Connect(WithLoading) />
</ContextProvider>
`;
